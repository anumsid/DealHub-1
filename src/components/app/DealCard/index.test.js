import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { DealCard } from './index'

describe('Deal Card Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      card: {
        id: 0,
        title: 'Cool Deal',
        votes: ['', ''],
        comments: ['', ''],
        image: '',
        description: 'Hello World',
        location: 'New York',
        category: 'Food & Beverages',
        expiration_date: '2018-07-31T00:00:00.000Z'
      }
    }
    wrapper = shallow(<DealCard {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
