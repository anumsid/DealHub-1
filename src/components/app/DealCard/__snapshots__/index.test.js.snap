// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal Card Component Matches snapshot 1`] = `
<div
  className="card mx-auto"
  style={
    Object {
      "width": "30rem",
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <Label
      color="pink"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <i
        aria-hidden={true}
        className="like icon"
      />
      2
    </Label>
    <Label
      className="ui label"
      color="blue"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <i
        aria-hidden={true}
        className="comment icon"
      />
      2
    </Label>
  </div>
  <Component
    card={
      Object {
        "category": "Food & Beverages",
        "comments": Array [
          "",
          "",
        ],
        "description": "Hello World",
        "expiration_date": "2018-07-31T00:00:00.000Z",
        "id": 0,
        "image": "",
        "location": "New York",
        "title": "Cool Deal",
        "votes": Array [
          "",
          "",
        ],
      }
    }
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    See Details
  </button>
</div>
`;
